### ISSUE TITLE
Lengths of arguments is not correctly checked

### ISSUE THREAD

**Issue by tdeenes** [mllg/checkmate#187]
In all of the `check*` functions that I tested, the arguments which require a length-1 object, can leak to C internals and produce wrong error messages:
```r
checkInteger(1:3, len = 1:2)
# Error in checkInteger(1:3, len = 1:2) : 
#  Argument '517cb964' must have length 1

checkDataFrame(data.frame(), ncols = 1:3)
# Error in isTRUE(lhs) : Argument '517cb986' must have length 1
```

**Comment by mllg** [comment]
Wrong placeholder in sprintf, thanks for reporting.
